### PR TITLE
allow to display Elissa from any view, make icon optional

### PR DESCRIPTION
**Pull request by jkandzi**

### CODE REVIEW

File: Elissa/Classes/Elissa.swift [hunk 1]
```diff
@@ -76,62 +58,94 @@ open class Elissa: UIView {
...
+        } else {
+            updatedFrame.size.width = messageLabel.frame.size.width + 16
+        }
         updatedFrame.size.height = popupHeight
         updatedFrame.origin.x = sourceView.center.x - updatedFrame.size.width / 2
-        updatedFrame.origin.y = (sourceView.frame.origin.y - sourceView.frame.size.height) + offsetToSourceView
+        updatedFrame.origin.y = (sourceView.frame.origin.y - popupHeight) - arrowSize.height + configuration.arrowOffset
+        
+        self.frame = updatedFrame
+        self.layer.cornerRadius = 5
```
Comment (Broich) [inline Elissa/Classes/Elissa.swift]: Are the `self`'s needed?

File: Elissa/Classes/Elissa.swift [hunk 2]
```diff
@@ -76,62 +58,94 @@ open class Elissa: UIView {
...
+    @IBAction func actionButtonTapped(_ sender: UIButton) {
+        completionHandler?()
+    }
+    
+    private func calculatePositon() {
+        guard let sourceView = sourceView else { return }
         var updatedFrame = CGRect()
-        updatedFrame.size.width = iconImageView.frame.size.width + messageLabel.frame.size.width + 24
+        if configuration.image != nil {
+            updatedFrame.size.width = iconImageView.frame.size.width + messageLabel.frame.size.width + 24
```
Comment (Broich) [inline Elissa/Classes/Elissa.swift]: Magic numbers - might be better to use a named constant?

File: Elissa/Classes/UIViewControllerExtension.swift [hunk 3]
```diff
@@ -37,13 +38,26 @@ extension UIViewController {
...
-        
         if let height = tabBarController?.tabBar.frame.size.height {
             sourceFrame.origin.y = UIScreen.main.applicationFrame.size.height - height / 2
         }
+        return UIView(frame: sourceFrame)
+    }
+}
+
+extension UIView {
+    public func showElissa(fromSourceView sourceView: UIView, configuration: ElissaConfiguration, onTouchHandler: (() -> Void)?) -> Elissa {
```
Comment (Broich) [inline Elissa/Classes/UIViewControllerExtension.swift:49]: Eventually add discardable result annotation? I had a warning when presenting.

**Review by Broich — APPROVED** [review]
Looks really cool 👍

One thing that might be a problem is that multiple `showElissa` calls create multiple instances and therefore subviews added to the parent view.

screen shot 2017-01-22 at 7 25 23 pm

Of course this is only important if the presentation of Elissa is not completely under your control :)

**Comment by Broich** [comment]
Let's make sure we create a new version in the trunk before merging ;)

**Comment by jkandzi** [comment]
In my use case I have to have multiple visible Elissas at the same time. So Elissa can't be a Singleton and multiple `showElissa` calls will create multiple Elissa instances.

That's also the reason why `func showElissa` does not have a `discardable result`. Here you have to keep the Elissa instance to be able to dismiss it later.

Maybe we have to add some documentation to make it clear that TabBar Elissas are Singletons and UIView Elissas are not?

**Comment by jkandzi** [comment]
Maybe also update the Example project?

**Comment by Broich** [comment]
Ok, I understand why a Singleton is not an option. Just wanted to make sure you're not surprised when multiple instances will be displayed in the same position on screen and tapping one won't remove all of them from their superview.

Updating the example project/documentation sounds like a good idea. Maybe as long as it is not stable you can go with

`pod 'Elissa', :git => "https://github.com/KitchenStories/Elissa", :branch =>  "3.0"`

**Pull request merged**